src/reducers/query.js

```diff
@@ -39,6 +39,9 @@ export default function (state = initialState, action) {
       return action.data.nextCursorMark ? {...state, cursorMark: action.data.nextCursorMark} : state;
     case "SET_GROUP":
       return {...state, group: action.group};
+    // new redux type that to set up new start and new searchFields
+    case "SET_INIT_STATE":
+        return {...state, searchFields: action.newFields, sortFields: action.newSortFields, start: state.pageStrategy === "paginate" ? action.newStart: null};
   }
```

Comment:
New redux reducer type to set up solrClient state.

Reply:
Same suggestion here -- it would be great to track our additions in a new file if possible.